workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs

```diff
@@ -70,25 +75,90 @@ private static void LaunchMobileClient()
                     .WithArgs("shell", "am", "start", "-S", "-n", $"{bundleId}/com.unity3d.player.UnityPlayerActivity",
                         "-e", "\"arguments\"", $"\\\"{arguments.ToString()}\\\"")
                     .Run();
-
-                EditorUtility.DisplayProgressBar("Launching Mobile Client", "Done", 1.0f);
             }
             finally
             {
                 EditorUtility.ClearProgressBar();
             }
         }
 
-        private static bool TryGetApkPath(string rootPath, out string apkPath)
+        [MenuItem(MenuLaunchiOSDevice, false, 74)]
+        private static void LaunchiOSDeviceClient()
         {
-            foreach (var file in Directory.GetFiles(rootPath, "*.apk", SearchOption.AllDirectories))
+            try
             {
-                apkPath = file;
-                return true;
-            }
+                // Ensure needed tools are installed
+                if (string.IsNullOrEmpty(LibIDeviceInstallerBinary))
+                {
+                    Debug.LogError("Could not find ideviceinstaller tool. Please ensure it is installed. " +
+                        "See https://github.com/libimobiledevice/ideviceinstaller fore more details.");
+                    return;
+                }
+
+                if (string.IsNullOrEmpty(LibIDeviceDebugBinary))
+                {
+                    Debug.LogError("Could not find idevicedebug tool. Please ensure libimobiledevice is installed. " +
+                        "See https://helpmanual.io/help/idevicedebug/ for more details.");
+                    return;
+                }
+
+                EditorUtility.DisplayProgressBar("Launching iOS Device Client", "Installing archive", 0.3f);
+
+                // Get chosen ios package id
+                var bundleId = PlayerSettings.GetApplicationIdentifier(BuildTargetGroup.iOS);
+                string appList = null;
+
+                // Find archive to install
+                var ipaPath = Directory.GetFiles(AbsoluteAppBuildPath, "*.ipa", SearchOption.AllDirectories).FirstOrDefault();
+                RedirectedProcess.Command(LibIDeviceInstallerBinary).WithArgs("-l")
+                    .AddAccumulatedOutputProcessing(output => { appList = output; })
+                    .RedirectOutputOptions(OutputRedirectBehaviour.None).Run();
+                var existsOnDevice = appList.Contains(bundleId);
+                var existsLocally = !string.IsNullOrEmpty(ipaPath);
+                if (!existsLocally && !existsOnDevice)
+                {
+                    Debug.LogError($"Could not find an app on device or built out iOS .ipa archive in \"{AbsoluteAppBuildPath}\" to launch.");
+                    return;
+                }
+
+                if (existsLocally)
```

Comment:
at this point we shouldn't need to check for it. It should be clear that the installation has succeeded and we can run the app

Reply:
See the comment about for the logic intended here.
The idea is that developer doesn't have to build .ipa every time they want to start the app with new config.